File: Naive_String_Matching/Naive_String_Matching.cpp
#include <iostream>

using namespace std;

void search(string text, string pattern)
{
    int lengthText = text.length();
    int lengthPattern = pattern.length();

    for(int i = 0; i <= lengthText - lengthPattern; i++)
    {
        for(int j = 0; j < lengthPattern; j++)
            if(text[i + j] != pattern[j])
                break;

        if(j == lengthPattern)
           cout << "Pattern found at " << i + 1 << endl;
    }
}

int main()
{
    cout << "Enter text" << endl;
    cin >> text;
    cout << "Enter pattern" << endl;
    search(text, pattern);
    return 0;
}

/* 
Input:
text = "namanchamanbomanamansanam"
pattern = "aman"

Output:
Pattern found at 2
Pattern found at 8
Pattern found at 17
*/

File: Naive_String_Matching/Naive_String_Matching.py
def search(text, pattern):
    lengthText = len(text)
    lengthPattern = len(pattern)

    for i in range(0, lengthText - lengthPattern):
        j = 0

        while j < lengthPattern:
            if text[i + j] != pattern[j]:
                break

            j += 1

        if j == lengthPattern:
            print("Pattern found at " + str(i + 1))

text = input()
pattern = input()
search(text, pattern)

'''
Input:
text = "namanchamanbomanamansanam"
pattern = "aman"

Output:
Pattern found at 2
Pattern found at 8
Pattern found at 17
'''
